HIP/hip-0000-introduction-of-blocks.md

```diff
@@ -0,0 +1,199 @@
+- [https://github.com/hashgraph/hedera-state-proof-verifier-go](https://github.com/hashgraph/hedera-state-proof-verifier-go)
+
+### Mirror Nodes
+
+Based on the updates specified above, the record stream objects will pass all the necessary information for Mirror Nodes to build `Record files`/`blocks` and store enough data about them to be able to answer all block related queries outlined above. To do this they will need to collect the `number`, `hash` and `timestamp` specified in the `Record file`. For old record files, they will be derived through a migration process as specified below.
+
+**Historical State**
+
+`Record files` prior to the introduction of the block properties will not expose the required information. In order to be able to answer historical queries, `mirror nodes` will need to iterate over the existing/already imported record files since genesis, assign the corresponding `block number`, `hash` and migrate their DB collections such that they are able to answer queries of type:
+
+- Getting block by hash/number
+- Getting a list of transactions for a block by number/hash
+- Getting logs based on block number filtering
+
+### Block Properties
+
+The following table specifies all `block` properties and at which point they will be computed. Mirror nodes must `hex` encode all properties prior to exposing them through their APIs. This table defines the properties that must be returned through APIs from Mirror Nodes.
+
+| Property | Computed By | Description  |
+| --- | --- | --- |
+| number | Services | Stored in services state (only the current number) and exported in the record stream. It is the consecutive number of the record file that is being incremented by 1 for every new record file. The number will be initially set in services through the bootstrapping process. It is exposed to the 1) EVM during Transaction Execution through the NUMBER opcode; 2) as a new property in the record file and ingested by mirror nodes. |
+| timestamp | Services | Stored in services state (only the last block timestamp) and computed by services. It is the consensusTimestamp of the first transaction in the record file. It is exposed to the 1) EVM during Transaction Execution through the TIMESTAMP opcode; 2) implicitly exported in the record file through the TS of the first transaction in the record file. |
+| hash | Services | Stored in services (last 256 blocks). It is the keccak256(runningHash) of the previous record file. That is the running hash of the last record stream object from the previous record file. It is exposed to the 1) EVM during Transaction Execution through the BLOCKHASH opcode; 2) In the record file as the End Object Running Hash |
+| baseFeePerGas (??) | Mirror Node | Computed by Mirror Node(s). It is a hex encoded value of the gas pricing based in the FeeSchedule file. |
```

Comment:
The meaning of this particular field however deals with EIP-1559 fee burn, which we don't do.
```suggestion
| baseFeePerGas | Mirror Node | Always zero, since there is no EIP-1559 style floating block capacity fees in Hedera. |
```